integration-tests/src/test/java/tachyon/worker/netty/protocol/RPCMessageIntegrationTest.java

```diff
@@ -0,0 +1,226 @@
+import io.netty.channel.socket.nio.NioSocketChannel;
+
+import tachyon.Constants;
+import tachyon.TestUtils;
+import tachyon.util.NetworkUtils;
+import tachyon.worker.netty.protocol.buffer.DataByteBuffer;
+import tachyon.worker.netty.protocol.buffer.DataFileChannel;
+
+/**
+ * This tests the encoding and decoding of RPCMessage's. This is done by setting up a simple
+ * client/server bootstrap connection, and writing messages on the client side, and verifying it on
+ * the server side. In this case, the server simply stores the message received, and does not reply
+ * to the client.
+ */
+public class RPCMessageIntegrationTest {
+  private static final long BLOCK_ID = 11;
+  private static final long OFFSET = 22;
+  private static final long LENGTH = 33;
+
+  // This channel initializer sets up a simple pipeline with the encoder and decoder.
+  private static class PipelineInitializer extends ChannelInitializer<SocketChannel> {
+    private MessageSavingHandler mHandler = null;
+
+    public PipelineInitializer(MessageSavingHandler handler) {
+      mHandler = handler;
+    }
+
+    @Override
+    protected void initChannel(SocketChannel ch) throws Exception {
+      ChannelPipeline pipeline = ch.pipeline();
+      pipeline.addLast("frameDecoder", RPCMessage.createFrameDecoder());
+      pipeline.addLast("RPCMessageDecoder", new RPCMessageDecoder());
+      pipeline.addLast("RPCMessageEncoder", new RPCMessageEncoder());
+      pipeline.addLast("handler", mHandler);
+    }
+  }
+
+  @Rule
+  public TemporaryFolder mFolder = new TemporaryFolder();
+
+  private Channel mOutgoingChannel;
+
+  private static NioEventLoopGroup sEventClient;
+  private static NioEventLoopGroup sEventServer;
+  private static MessageSavingHandler sIncomingHandler;
+  private static Bootstrap sBootstrapClient;
+  private static SocketAddress sLocalAddress;
+
+  @BeforeClass
+  public static void beforeClass() {
+    sEventClient = new NioEventLoopGroup(1);
+    sEventServer = new NioEventLoopGroup(1);
+    sIncomingHandler = new MessageSavingHandler();
+
+    // Setup the server.
+    ServerBootstrap bootstrap = new ServerBootstrap();
+    bootstrap.group(sEventServer);
+    bootstrap.channel(NioServerSocketChannel.class);
+    bootstrap.childHandler(new PipelineInitializer(sIncomingHandler));
+
+    InetSocketAddress address =
+        new InetSocketAddress(NetworkUtils.getLocalHostName(100), Constants.DEFAULT_MASTER_PORT);
+    ChannelFuture cf = bootstrap.bind(address).syncUninterruptibly();
+    sLocalAddress = cf.channel().localAddress();
+
+    // Setup the client.
+    sBootstrapClient = new Bootstrap();
+    sBootstrapClient.group(sEventClient);
+    sBootstrapClient.channel(NioSocketChannel.class);
+    sBootstrapClient.handler(new PipelineInitializer(new MessageSavingHandler()));
+  }
+
+  @AfterClass
+  public static void afterClass() {
+    // Shut everything down.
+    sEventClient.shutdownGracefully();
```

Comment:
I vaguely remember @calvinjia had PRs before to change shutdownGracefully() to shutdown() before to speed up the tests where we don't need to be graceful :))) Please double check with him if we should do the samething here.